corehq/apps/hqadmin/views.py

```diff
@@ -263,36 +265,43 @@ def _to_chart_data(data_dict):
 
 @require_superuser_or_developer
 def mass_email(request):
-    if not request.couch_user.is_staff:
-        raise Http404()
-
     if request.method == "POST":
         form = EmailForm(request.POST)
         if form.is_valid():
             subject = form.cleaned_data['email_subject']
-            body_html = form.cleaned_data['email_body_html']
-            body_text = form.cleaned_data['email_body_text']
+            body_html = Template(form.cleaned_data['email_body_html'])
+            body_text = Template(form.cleaned_data['email_body_text'])
             real_email = form.cleaned_data['real_email']
 
             if real_email:
-                recipients = [h['username'] for h in UserES().web_users().run().hits]
+                recipients = [{
+                    'username': h['username'],
+                    'first_name': h['first_name'] or 'CommCare User',
+                } for h in UserES().web_users().run().hits]
             else:
-                recipients = [request.couch_user.username]
+                recipients = [{
+                    'username': request.couch_user.username,
+                    'first_name': request.couch_user.first_name or 'CommCare User',
+                }]
 
             for recipient in recipients:
+                context = recipient
+                context.update({
+                    'url_prefix': '' if settings.STATIC_CDN else 'http://' + get_site_domain(),
```

Comment:
why `http`?

Reply:
Copied & pasted from `registration/utils.py` Does it seem problematic?